src/lib.rs

```diff
@@ -217,12 +212,37 @@ pub struct VirtualDestination {
 #[derive(PartialEq)]
 pub struct Device { object: Object }
 
+#[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+type AlignmentMarker = [u32; 0]; // ensures 4-byte alignment (on ARM)
+
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+type AlignmentMarker = [u8; 0]; // unaligned
+
 /// A [list of MIDI events](https://developer.apple.com/reference/coremidi/midipacketlist) being received from, or being sent to, one endpoint.
 ///
-#[derive(PartialEq)]
-pub struct PacketList(*const MIDIPacketList);
+#[repr(C)]
+pub struct PacketList {
+    // NOTE: This type must only exist in the form of immutable references
+    //       pointing to valid instances of MIDIPacketList.
+    //       This type must NOT implement `Copy`!
+    inner: PacketListInner,
+    _do_not_construct: AlignmentMarker
```

Comment:
is `_do_not_construct` a convention name for such cases ? If not, could you rename to something like `_alignment_marker` ?

Reply:
It's not a convention. I just did this to solve two problems at once, namely (1) to have the correct alignment and (2) to give a strong hint that any code that would try to construct an instance of this type would be wrong (but this is only relevant within the library, not users of the library). I can rename if you prefer.

Reply:
Not feeling strong about this renaming, but given that it is not exposed to the public through the API, (1) is the main point to hint, so the proposed name looks more intuitive to me. You can focus on other comments first, and if having enough time do this ;-)